### PR TITLE
Enable namespaced broker again

### PR DESCRIPTION
**Pull request by aliok**
Fixes #

## Proposed Changes

- Enable namespaced broker again
- Adjust tests for the new validation rules per webhook (no cross namespace spec.config)
- 
## **Blocked by:**
- [x] https://github.com/knative/eventing/pull/6510
- [x] https://github.com/knative/test-infra/pull/3514
- [x] https://github.com/knative-sandbox/eventing-kafka-broker/pull/2570 (not 100% sure)
- [x] Upgrading eventing dependency: https://github.com/knative-sandbox/eventing-kafka-broker/pull/2610
- [x] https://github.com/knative-sandbox/eventing-kafka-broker/pull/2610 requires https://github.com/knative-sandbox/eventing-kafka-broker/pull/2609

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: test/e2e-tests.sh
Comment (matzew) [inline test/e2e-tests.sh:40]: I think you want to revert this temporary fix too:
https://github.com/knative-sandbox/eventing-kafka-broker/pull/2560

**Comment by aliok** [comment]
I ran `e2e_broker` and `e2e_new` tests locally.

Like this:
```
export SYSTEM_NAMESPACE=knative-eventing
export BROKER_CLASS=KafkaNamespaced
go test /Users/aliok/go/src/knative.dev/eventing-kafka-broker/test/e2e_new/... -tags=e2e -count=1 -race -timeout=1h
```

Everything is green locally.

File: test/e2e_new/broker_conformance_test.go
Comment (aliok) [inline test/e2e_new/broker_conformance_test.go:50]: There are limitations in broker conformance tests upstream. This part needs a little work, possibly in the upstream to make the test cases extensible/decoratable.

I still need to figure this part out.

File: vendor/knative.dev/eventing/test/rekt/features/broker/control_plane.go
Comment (aliok) [inline vendor/knative.dev/eventing/test/rekt/features/broker/control_plane.go:51]: There are some limitations in the upstream broker conformance tests. As a quick hack, I edited the files in vendor directory. Once PR passes fine, I will create a PR in knative/eventing for addressing those limitations.

**Comment by aliok** [comment]
/test reconciler-tests_eventing-kafka-broker_main
/test integration-tests_eventing-kafka-broker_main

Build tests are failing and that's fine for now. See the PR desc

File: vendor/knative.dev/eventing/test/rekt/features/broker/control_plane.go
Comment (aliok) [reply vendor/knative.dev/eventing/test/rekt/features/broker/control_plane.go:51]: Created https://github.com/knative/eventing/pull/6510

File: control-plane/cmd/kafka-controller/main.go
Comment (matzew) [inline control-plane/cmd/kafka-controller/main.go:81]: I'd still like to see them called `namespaced-xxx-controller`

File: test/e2e_broker/conformance/control_plane_conformance_test.go
Comment (matzew) [inline test/e2e_broker/conformance/control_plane_conformance_test.go:36]: the `GetKafkaClassFromEnv` is now inlined to `CreatorWithOption`, right? 

I am wondering why to was before explicit on a few tests here 
(Not saying we need to factor out this change, just curious)

File: control-plane/cmd/kafka-controller/main.go
Comment (aliok) [reply control-plane/cmd/kafka-controller/main.go:81]: Yeah, right... I forgot that. I cross my heart I will do that renaming :)

File: test/e2e_broker/conformance/control_plane_conformance_test.go
Comment (aliok) [reply test/e2e_broker/conformance/control_plane_conformance_test.go:36]: There's no big reason other than me not seeing there's a better, unified and reusable way.

**Comment by aliok** [comment]
/close

I don't get why BROKER_CLASS is missing in some tests. Gonna reopen.

**Comment by aliok** [comment]
/test integration-tests-namespaced-broker_eventing-kafka-broker_main

```
failed to sync configmap cache: timed out waiting for the condition
``` :(

**Comment by aliok** [comment]
/test integration-tests-namespaced-broker_eventing-kafka-broker_main

Trying one last time before debugging locally.

File: test/pkg/broker/broker.go
Comment (pierDipi) [inline test/pkg/broker/broker.go:80]: what's the problem with this?

File: test/e2e_new/features/featuressteps/broker.go
Comment (pierDipi) [inline test/e2e_new/features/featuressteps/broker.go:75]: Can we use the constant we have?

File: test/e2e_new/features/broker_deleted_recreated.go
Comment (pierDipi) [inline test/e2e_new/features/broker_deleted_recreated.go:40]: Can we use the constant we have?

File: test/e2e_new/dls_test.go
Comment (pierDipi) [inline test/e2e_new/dls_test.go:211]: Can we use the constant we have?

File: test/e2e_new/dls_test.go
Comment (pierDipi) [inline test/e2e_new/dls_test.go:141]: Can we use the constant we have?

File: test/e2e_new/dls_test.go
Comment (pierDipi) [inline test/e2e_new/dls_test.go:75]: Can we use the constant we have?

Also, can we extract this setup into a separate function to avoid duplication?

File: test/e2e_new/tracing_test.go
Comment (pierDipi) [inline test/e2e_new/tracing_test.go:80]: Can we use the constant we have?

File: test/e2e_new/tracing_test.go
Comment (pierDipi) [inline test/e2e_new/tracing_test.go:267]: Can we use the constant we have?

**Comment by pierDipi** [comment]
Constants are here: https://github.com/knative-sandbox/eventing-kafka-broker/blob/37c43784d7144fb343aaa3a3c74a437b60e06f06/test/pkg/run.go#L29-L32

**Comment by aliok** [comment]
@pierDipi thanks for the review. I would like to address your comments in a separate PR.

This PR is waiting for too long and I am tired of rebasing because of the parallel changes. Let's merge this and I will address your comments.

So, please lgtm if all good.

**Comment by aliok** [comment]
/test reconciler-tests-namespaced-broker_eventing-kafka-broker_main

Failing at `TestBrokerConfigMapDoesNotExist` but passes locally. Retesting

**Comment by aliok** [comment]
> Failing at TestBrokerConfigMapDoesNotExist but passes locally. Retesting

Passes locally when I run the test alone. It fails when I run the whole suite 🤷

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Comment by matzew** [comment]
```
    --- FAIL: TestBrokerConfigMapDeletedFirst/delete_broker_ConfigMap_first/Assert/[Stable/MUST]delete_broker (120.22s)
```

/retest

**Comment by matzew** [comment]
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-shade-plugin:3.3.0:shade (default) on project receiver: Execution default of goal org.apache.maven.plugins:maven-shade-plugin:3.3.0:shade failed: Plugin org.apache.maven.plugins:maven-shade-plugin:3.3.0 or one of its dependencies could not be resolved: Could not transfer artifact org.apache.maven.shared:maven-artifact-transfer:jar:0.13.1 from/to central (https://repo.maven.apache.org/maven2): transfer failed for https://repo.maven.apache.org/maven2/org/apache/maven/shared/maven-artifact-transfer/0.13.1/maven-artifact-transfer-0.13.1.jar: Connection reset -> [Help 1]
[ERROR] 
```

/retest

**Comment by matzew** [comment]
```
        	Error:      	Expected nil, but got: &errors.errorString{s:"timed out waiting for the condition"}
```

Again, there is something w/ the `TestBrokerConfigMapDeletedFirst` :disappointed:

**Comment by aliok** [comment]
> Again, there is something w/ the TestBrokerConfigMapDeletedFirst 😞

Yeah, there's some flakiness. Also in the similar tests like `TestBrokerConfigMapDoesNotExist` and `TestBrokerDeletedRecreated`.

I try locally and everything is fine. Gonna create a test PR which disables all tests except these to see what happens.

**Comment by matzew** [comment]
Locally that specific test worked fine. Executed it (explicitly) a few times this morning (with both broker classes)...

**Comment by matzew** [comment]
/lgtm

/hold
Feel free unold... I saw passing before, and locally I got those tests also working _fine

However, there is a slight risk/fear of flakes.... If so, we need to continue to improve this.

**Comment by matzew** [comment]
@aliok I think there is a bit more going on, on the "namespaced" reconciler test :thinking:

**Comment by aliok** [comment]
> @aliok I think there is a bit more going on, on the "namespaced" reconciler test 🤔

But what? 🤷 

I am trying the same suite locally, everything is good. I increased the cluster size as well in the test script.


/test reconciler-tests-namespaced-broker

**Comment by aliok** [comment]
Making some experiments with timeouts, adding some log statements, etc here: https://github.com/knative-sandbox/eventing-kafka-broker/pull/2646

**Review by matzew — APPROVED** [review]
/approve

/lgtm

**Pull request merged**
